### PR TITLE
Refactor ToLaurelTest: concise signature builders for readability

### PR DESCRIPTION
**Pull request by tautschnig**
Replace verbose programmatic AST construction (mkFuncSig, identType, mkArg, etc.) with concise builders that read more naturally:

  Before: mkFuncSig "f" (identType .builtinsInt)
            (args := #[mkArg "x" (identType .builtinsStr)])
  After:  func "f" int (args := #[arg "x" str])

Type shorthands: str, int, bool_, float_, bytes, any, none_, list_, dict_
Signature builders: func, overload, classDef, method, typeDef, externType
Argument helpers: arg, optArg

Also reorganizes the file: formatting functions before test runners, removes duplicate sections, and adds section headers.

Addresses MikaelMayer's feedback on PR #809 about test maintainability.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: StrataTest/Languages/Python/ToLaurelTest.lean [hunk 1]
```diff
@@ -94,7 +76,105 @@ private def runTestErrors (sigs : Array Signature) (modulePrefix : String := "")
...
+    args := { args := #[arg "self" str] ++ args, kwonly := #[] }
+    returnType := ret
+    isOverload := false
+    preconditions := #[], postconditions := #[] }
+
+private def typeDef (name : String) (definition : SpecType) : Signature :=
+  .typeDef { loc, nameLoc := loc, name, definition }
+
+private def externType (name : String) (ident : PythonIdent) : Signature :=
+  .externTypeDecl name ident
```
Comment (atomb) [inline StrataTest/Languages/Python/ToLaurelTest.lean:192]: I was just commenting on #818 that we could use some more abstractions for building Laurel code, to use in the translator, as well. I wonder whether these definitions would be more useful if they were part of the translator proper, rather than `private` `def`s used only in this test.

Comment (tautschnig) [reply StrataTest/Languages/Python/ToLaurelTest.lean:192]: See #853 for the Laurel translator - the ones here are for PySpec and could be confusing if re-used elsewhere.

**Comment by tautschnig** [comment]
@keyboardDrummer-bot Please resolve git conflicts.

**Comment by tautschnig** [comment]
@keyboardDrummer-bot Please resolve git conflicts.

**Review by joehendrix — APPROVED** [review]

**Review by atomb — APPROVED** [review]
